apps/web/src/hooks/usePermitAllowance.ts

```diff
@@ -0,0 +1,106 @@
+import { useTranslation } from '@pancakeswap/localization'
+import {
+  AllowanceTransfer,
+  generatePermitTypedData,
+  PERMIT2_ADDRESS,
+  PermitSingle,
+  Permit2ABI,
+} from '@pancakeswap/permit2-sdk'
+import { Permit2Signature } from '@pancakeswap/universal-router-sdk'
+import { CurrencyAmount, Token } from '@pancakeswap/swap-sdk-core'
+import { SLOW_INTERVAL } from 'config/constants'
+import { useCallback, useMemo } from 'react'
+import { isUserRejected } from 'utils/sentry'
+import { transactionErrorToUserReadableMessage } from 'utils/transactionErrorToUserReadableMessage'
+import { publicClient } from 'utils/wagmi'
+import { TransactionRejectedError } from 'views/Swap/V3Swap/hooks/useSendSwapTransaction'
+import { useQuery, useSignTypedData } from 'wagmi'
+import { Address, zeroAddress } from 'viem'
+import useAccountActiveChain from './useAccountActiveChain'
+import { useActiveChainId } from './useActiveChainId'
+
+export function usePermitAllowance(token?: Token, owner?: Address, spender?: Address) {
+  const { chainId } = useActiveChainId()
+
+  const inputs = useMemo<[Address, Address, Address]>(
+    () => [owner ?? zeroAddress, token?.address ?? zeroAddress, spender ?? zeroAddress],
+    [owner, spender, token?.address],
+  )
+
+  const { data: allowance } = useQuery(
+    [chainId, token?.address, owner, spender],
```

Comment:
let's add an identity prefix for this query key

Reply:
yup